python/cugraph/structure/graph_implementation/simpleGraph.py

```diff
@@ -0,0 +1,838 @@
+            The gdf column contains the weight value for each edge.
+            The expected type of the gdf_column element is floating point
+            number.
+        """
+
+        if self.adjlist is None:
+            if self.transposedadjlist is not None and\
+             self.properties.directed is False:
+                off, ind, vals = (
+                    self.transposedadjlist.offsets,
+                    self.transposedadjlist.indices,
+                    self.transposedadjlist.weights,
+                )
+            else:
+                off, ind, vals = graph_primtypes_wrapper.view_adj_list(self)
+            self.adjlist = self.AdjList(off, ind, vals)
+
+            if self.batch_enabled:
+                self._replicate_adjlist()
+
+        return self.adjlist.offsets, self.adjlist.indices, self.adjlist.weights
+
+    def view_transposed_adj_list(self):
+        """
+        Display the transposed adjacency list. Compute it if needed.
+        Returns
+        -------
+        offset_col : cudf.Series
+            This cudf.Series wraps a gdf_column of size V + 1 (V: number of
+            vertices).
+            The gdf column contains the offsets for the vertices in this graph.
+            Offsets are in the range [0, E] (E: number of edges).
+        index_col : cudf.Series
+            This cudf.Series wraps a gdf_column of size E (E: number of edges).
+            The gdf column contains the destination index for each edge.
+            Destination indices are in the range [0, V) (V: number of
+            vertices).
+        value_col : cudf.Series or ``None``
+            This pointer is ``None`` for unweighted graphs.
+            For weighted graphs, this cudf.Series wraps a gdf_column of size E
+            (E: number of edges).
+            The gdf column contains the weight value for each edge.
+            The expected type of the gdf_column element is floating point
+            number.
+        """
+
+        if self.transposedadjlist is None:
+            if self.adjlist is not None and self.properties.directed is False:
+                off, ind, vals = (
+                    self.adjlist.offsets,
+                    self.adjlist.indices,
+                    self.adjlist.weights,
+                )
+            else:
+                (
+                    off,
+                    ind,
+                    vals,
+                ) = graph_primtypes_wrapper.view_transposed_adj_list(self)
+            self.transposedadjlist = self.transposedAdjList(off, ind, vals)
+
+            if self.batch_enabled:
+                self._replicate_transposed_adjlist()
+
+        return (
+            self.transposedadjlist.offsets,
+            self.transposedadjlist.indices,
+            self.transposedadjlist.weights,
+        )
+
+    def delete_adj_list(self):
+        """
+        Delete the adjacency list.
+        """
+        self.adjlist = None
+
+    def enable_batch(self):
```

Comment:
is SimpleGraphImpl supposed to support batch long-term, or should this be on the distributed version of this class?

Reply:
Will be addressed when we look into updating batch functionalities in 0.20

Reply:
added fixme